### ISSUE TITLE
Crashes when encountering regex capture group

### ISSUE THREAD

**Issue by piotaixr** [containous/traefik#2272]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I updated from 1.3.x to 1.4.0

### What did you expect to see?

Traefik to reboot and serve content as usual

### What did you see instead?

An error in the log:

```
time="2017-10-17T03:49:00Z" level=info msg="Using TOML configuration file /etc/traefik/config.toml" 
time="2017-10-17T03:49:02Z" level=error msg="Error in Go routine: route {subdomain:(.+)?}.xxxxxxxx.com contains capture groups in its regexp. Only non-capturing groups are accepted: e.g. (?:pattern) instead of (pattern)" 
goroutine 69 [running]:
runtime/debug.Stack(0x69470f, 0xc420086280, 0x1f7402d)
       /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
       /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1b13d20, 0xc4202b1e50)
       /go/src/github.com/containous/traefik/safe/routine.go:148 +0x7d
github.com/containous/traefik/safe.GoWithRecover.func1.1(0x1feb478)
       /go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
panic(0x1b13d20, 0xc4202b1e50)
       /usr/local/go/src/runtime/panic.go:491 +0x283
github.com/containous/traefik/vendor/github.com/containous/mux.newRouteRegexp(0xc420ac6a20, 0x1f, 0x1, 0x2f38960, 0x20020a06290, 0x8)
       /go/src/github.com/containous/traefik/vendor/github.com/containous/mux/regexp.go:115 +0x11e2
github.com/containous/traefik/vendor/github.com/containous/mux.(*Route).addRegexpMatcher(0xc420ae6cf0, 0xc420ac6a20, 0x1f, 0x1000001, 0xc4200849e0, 0xc420ae6cf0)
       /go/src/github.com/containous/traefik/vendor/github.com/containous/mux/route.go:180 +0xd0
github.com/containous/traefik/vendor/github.com/containous/mux.(*Route).Host(0xc420ae6cf0, 0xc420ac6a20, 0x1f, 0x0)
       /go/src/github.com/containous/traefik/vendor/github.com/containous/mux/route.go:284 +0x4f
github.com/containous/traefik/vendor/github.com/containous/mux.(*Router).Host(0xc420594f50, 0xc420ac6a20, 0x1f, 0x1f)
       /go/src/github.com/containous/traefik/vendor/github.com/containous/mux/mux.go:247 +0xc8
github.com/containous/traefik/server.(*Rules).hostRegexp(0xc4205808e0, 0xc420580940, 0x2, 0x2, 0xc4209d0e70)
       /go/src/github.com/containous/traefik/server/rules.go:41 +0xd5
github.com/containous/traefik/server.(*Rules).(github.com/containous/traefik/server.hostRegexp)-fm(0xc420580940, 0x2, 0x2, 0x0)
       /go/src/github.com/containous/traefik/server/rules.go:146 +0x48
reflect.Value.call(0x1b58b80, 0xc4202b1b60, 0x13, 0x1f57c4c, 0x4, 0xc420aee540, 0x1, 0x2, 0x1f40701, 0x1b58b80, ...)
       /usr/local/go/src/reflect/value.go:434 +0x906
reflect.Value.Call(0x1b58b80, 0xc4202b1b60, 0x13, 0xc420aee540, 0x2, 0x2, 0xc, 0x0, 0xc420a067e0)
       /usr/local/go/src/reflect/value.go:302 +0xa4
github.com/containous/traefik/server.(*Rules).Parse.func1(0xc420be3000, 0xa, 0x1b58b80, 0xc4202b1b60, 0xc420580900, 0x2, 0x2, 0x2, 0xc4205ba000)
       /go/src/github.com/containous/traefik/server/rules.go:219 +0x1ed
github.com/containous/traefik/server.(*Rules).parseRules(0xc4205808e0, 0xc420be3000, 0x38, 0xc420a06a68, 0x50, 0x58)
       /go/src/github.com/containous/traefik/server/rules.go:201 +0xb1c
github.com/containous/traefik/server.(*Rules).Parse(0xc4205808e0, 0xc420be3000, 0x38, 0x410f38, 0xc420a077e8, 0xc420a077e0)
       /go/src/github.com/containous/traefik/server/rules.go:212 +0x77
github.com/containous/traefik/server.getRoute(0xc420ae8fc0, 0xc420a07018, 0xc420a077c0, 0xc420ae6c60)
       /go/src/github.com/containous/traefik/server/server.go:1147 +0x77
github.com/containous/traefik/server.(*Server).loadConfig(0xc42021d000, 0xc4209d0cc0, 0x2540be400, 0x100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1f588a3, ...)
       /go/src/github.com/containous/traefik/server/server.go:770 +0x401
github.com/containous/traefik/server.(*Server).listenConfigurations(0xc42021d000, 0xc420469490)
       /go/src/github.com/containous/traefik/server/server.go:388 +0x170
github.com/containous/traefik/server.(*Server).Start.func2(0xc420469490)
       /go/src/github.com/containous/traefik/server/server.go:188 +0x34
github.com/containous/traefik/safe.(*Pool).Go.func1()
       /go/src/github.com/containous/traefik/safe/routine.go:78 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x1feb478, 0xc42096df50)
       /go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
       /go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```
Removing the unused capture group fixes the problem but as 1.4.0 is a minor version, i was not expecting any breaking changes

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0
Codename:     roquefort
Go version:   go1.9.1
Built:        2017-10-16_04:52:07PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Provider: docker
Everything else mostly defaults.

**Comment by ldez** [comment]
Thanks for reporting.

The problem is due to https://github.com/gorilla/mux, a breaking change has been added in https://github.com/gorilla/mux/pull/218.

The current workaround: replace `(foo)?` by `(?:foo)?`

**Comment by traefiker** [comment]
Closed by #2296.
